### PR TITLE
[SDS-220] add map visualization functions

### PR DESCRIPTION
**Pull request by suzinyou**
Reviewer: @lickem22 

Estimate: 20min

---

## Ticket

Fixes: https://idinsight.atlassian.net/browse/SDS-220

### Goal

Provide map visualization functions for viewing
1. Distribution of all enumerators and targets
2. Assignment of targets to enumerators

### Changes

1. Move visualization functions from example pipeline to code
2. Use those functions in example_pipeline.ipynb
3. Add simple unit tests

### Future Tasks (optional)

We are using Folium icon compatible set of colours. There are only 19. This can be limiting in some situation if there are more than 19 enumerators are in one region.

## How has this been tested?

- Run tests
- Run example_pipeline.ipynb

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [x] I have updated the automated tests (if applicable)
- [x] I have updated the requirements (if applicable)
- [ ] I have updated the README file (if applicable)
- [x] I have updated affected documentation (if applicable)

### CODE REVIEW

File: surveyscout/visualize.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        "lightred",
+        "darkred",
+        "green",
+        "white",
+        "lightblue",
+        "gray",
+    ]
+    if len(enum_locations) > len(colors):
+        raise ValueError(
+            f"Too many enumerators to plot: {len(enum_locations)}. Currently, "
```
Comment (lickem22) [inline surveyscout/visualize.py:82]: Maybe we should warn a warning and only display the max number of enumerators we can?

File: surveyscout/visualize.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    enum_ids = enum_locations.get_ids()
+    color_map = dict(zip(enum_ids, [color for color in colors]))
+
+    # Create map group for each enumerator
+    groups = {enum_id: folium.FeatureGroup(name=enum_id) for enum_id in enum_ids}
+
+    # Merge target locations to assignments table
+    result_with_locs = assignments.merge(
+        target_locations.get_df(),
+        left_on="target_id",
```
Comment (lickem22) [inline surveyscout/visualize.py:95]: Why did you use `target_id`for `left_on` but `target_locations.get_id_column()` for `right_on`?

**Review by lickem22 — APPROVED** [review]
Thanks!

File: surveyscout/visualize.py (hunk 1 above)
Comment (suzinyou) [reply surveyscout/visualize.py:82]: Good idea!

File: surveyscout/visualize.py (hunk 2 above)
Comment (suzinyou) [reply surveyscout/visualize.py:95]: Ahhh thanks for catching

**Pull request merged**